### DISCUSSION TITLE
how can i style an element which is not belong to common state like parent siblings

### DISCUSSION THREAD

**Discussion by codenoobforreal** [tailwindlabs/tailwindcss#13043] [Help] [answered]
This is my first time to use tailwind in my react project.
here's the html structure of my component:
```html
<label>
    <input>
    <div>
        <span>style this span element when input is checked and label has specified classname for example "tailwind"< /span>
    </div>
</label>
```
the relation between input and span is not mentioned in https://tailwindcss.com/docs/hover-focus-and-other-states#pseudo-classes.
btw,the "tailwind" classname is passed by props.

**Comment by alex-krasikau** [comment]
Hey @codenoobforreal!

Thank you for your question. I suggest asking more real-world questions to help us understand your situation better and provide more useful advice.

Discussing your scenario specifically, a direct but not recommended approach is to utilize CSS selectors to apply styles according to the input state:
```html
<label class="[&.tailwind>input:checked+div>span]:font-bold">
  <input type="checkbox" />
  <div>
    <span>style this span element when input is checked and label has specified classname for example "tailwind"</span>
  </div>
</label>
```

You can check out the full example here - https://play.tailwindcss.com/iWG6fFxD5W.

When using React, it's advisable to utilize state variables and template strings to address the problem:
```jsx
function MyComponent({ className }) {
  const [isChecked, setIsChecked] = React.useState(false);

  return (
    <label className={className}>
      <input type="checkbox" onChange={() => setIsChecked(prev => !prev)} />
        <div>
          <span className={`transition-colors ${isChecked && className === 'tailwind' ? 'text-red-500' : 'text-black'}`}>
            Style this span element when input is checked and label has specified classname for example "tailwind"
          </span>
        </div>
    </label>
  );
}
```

You can enhance this solution further by utilizing a library such as https://github.com/lukeed/clsx, which offers a simpler way to construct classes.

**Reply by codenoobforreal** [reply]
Hey @alex-krasikau!
Thank you for the solutions and advices!
These two approaches did solve the problem,the first one used css selectors directly and the second one added a input state.
I wonder if there's a better way that no extra react state and css selectors?A way that use something like "group" and "peer" in case I missed some useful doc.

**Reply by alex-krasikau** [reply]
Yup, here you go - https://play.tailwindcss.com/8ao07nJYfL

```html
<label class="tailwind group">
  <input class="peer" type="checkbox" />
  <span class="group-[.tailwind]:peer-checked:font-bold">Label</span>
</label>
```

**Reply by codenoobforreal** [reply]
I know this solution when span is input's sibling.
the origin one is
```html
<label>
    <input>
    <div>
        <span>peer not working in this scenario</span>
    </div>
</label>
```

**Reply by alex-krasikau** [reply]
Here's the solution - https://play.tailwindcss.com/VTHwWroCq4

```html
<label class="tailwind group">
  <input class="peer" type="checkbox" />
  <div class="group-[.tailwind]:peer-checked:*:font-bold">
    <span class="">Label</span>
  </div>
</label>

```

**Reply by codenoobforreal** [reply]
nice one.
That remind me to a more common scenario.
When the span element or let's say target element is not the direct child.
```html
<label>
    <input>
    <div classname="outter">
        <div>
           <span>not direct child of outter div</span>
       </div>
    </div>
</label>
```
also this one.
I just want to style span element not the p element.
```html
<label class="tailwind group">
  <input class="peer" type="checkbox" />
  <div class="group-[.tailwind]:peer-checked:*:font-bold">
    <p>not me</p>
    <span class="">just me</span>
  </div>
</label>
```

**Reply by alex-krasikau** [reply, marked as answer]
https://play.tailwindcss.com/05FTBduYlP

```html
<div class="p-8">
  <label class="tailwind group flex gap-x-2">
    <input class="peer" type="checkbox" />
    <div class="group-[.tailwind]:peer-checked:[&_span]:font-bold">
      <div>
        <span>not direct child of outter div</span>
      </div>
    </div>
  </label>
</div>

<div class="p-8">
  <label class="tailwind group flex gap-x-2">
    <input class="peer" type="checkbox" />
    <div class="group-[.tailwind]:peer-checked:[&_span]:font-bold">
      <p>not me</p>
      <span class="">just me</span>
    </div>
  </label>
</div>

```

**Reply by codenoobforreal** [reply]
ty!
that's the key part I missunderstood in the document.
https://tailwindcss.com/docs/hover-focus-and-other-states#custom-modifiers
combining these modifies expecially custom modifier can handle this situation.
